### ISSUE TITLE
Small compilation issues on NetBSD

### ISSUE THREAD

**Issue by Rhialto** [bictorv/chaosnet-bridge#37]
Hi! I was just compiling this (or the version included as a submodule in the its repo) on NetBSD and had to make some trivial changes:

```
diff --git a/cbridge.h b/cbridge.h
index 8fa2d79..1bed6f7 100644
--- a/cbridge.h
+++ b/cbridge.h
@@ -37,7 +37,7 @@
 #if CHAOS_ETHERP
 #ifndef ETHER_BPF
 // use BPF rather than sockaddr_ll, e.g. for MacOS
-#if __APPLE__ || __OpenBSD__
+#if __APPLE__ || __OpenBSD__ || __NetBSD__
 #define ETHER_BPF 1
 #else
 #define ETHER_BPF 0
@@ -71,7 +71,7 @@
 #include <netinet/in.h>
 #include <net/if.h>
 #include <net/if_arp.h>
-#if __OpenBSD__
+#if __OpenBSD__ || __NetBSD__
 #include <netinet/if_ether.h>
 #else
 #include <net/ethernet.h>
```
Also there was a warning from the compiler:
```
.../its/tools/cbridge$ gmake
cc -c -g -o chtls.o chtls.c
chtls.c: In function 'forward_on_tls':
chtls.c:1439:25: warning: passing argument 2 of 'is_in_mux_list' from incompatible pointer type [-Wincompatible-pointer-types]
   is_in_mux_list(dchad, &tlsdest[i].tls_muxed)
                         ^
chtls.c:1404:1: note: expected 'u_short * {aka short unsigned int *}' but argument is of type 'u_short (*)[4] {aka short unsigned int (*)[4]}'
 is_in_mux_list(u_short addr, u_short *list)
 ^~~~~~~~~~~~~~
```
which was easily fixed by
```diff --git a/chtls.c b/chtls.c
index 954dae8..9a595d1 100644
--- a/chtls.c
+++ b/chtls.c
@@ -1436,7 +1436,7 @@ forward_on_tls(struct chroute *rt, u_short schad, u_short dchad, struct chaos_he
         (rt->rt_braddr == 0 && (tlsdest[i].tls_addr == rt->rt_dest))
         ||
         // multiplexed
-        is_in_mux_list(dchad, &tlsdest[i].tls_muxed)
+        is_in_mux_list(dchad, &tlsdest[i].tls_muxed[0])
         )) {
       if (verbose || debug) fprintf(stderr,"Forward TLS to dest %#o over %#o (%s)\n", dchad, tlsdest[i].tls_addr, tlsdest[i].tls_name);
       td = &tlsdest[i];
```
According to the declaration `u_short tls_muxed[CHTLS_MAXMUX]`, to get a pointer to a short you indeed need to get the address of `tls_muxed[0]`.
I guess gcc has become pickier when decaying an array to the address of the first element.

**Comment by bictorv** [comment]
Thanks! The fixes for cbridge.h will be applied, but the warning for chtls.c was already handled by Lars' by removing the &.

**Comment by bictorv** [comment]
Fixed by commit #8962055
